src/NuGet.ContentModel/ContentPropertyDefinition.cs

```diff
@@ -146,6 +157,11 @@ public virtual bool TryLookup(string name, out object value)
 
         public Func<object, object, bool> CompatibilityTest { get; }
 
+        /// <summary>
+        /// Find the nearest compatible candidate.
+        /// </summary>
+        public Func<object, object, object, int> CompareTest { get; }
```

Comment:
this looks like a bug farm waiting to happen, and so do all the method calls with a million nulls. Can we track cleanup for U1